src/logger.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { format } from 'node:util';
+import Interfaces from '@oclif/core/interfaces';
+import { Logger } from '@salesforce/core';
+
+export const customLogger = (namespace: string): Interfaces.Logger => {
+  const sfLogger = new Logger(namespace);
+  return {
+    child: (ns: string, delimiter?: string) => customLogger(`${namespace}${delimiter ?? ':'}${ns}`),
+    debug: (formatter: unknown, ...args: unknown[]) => sfLogger.debug(format(formatter, ...args)),
```

Comment:
I was recently bit by pulling methods off of the underlying pino logger like this...it turned into some `unbound this` problem.  Have you tested this with your CLI ?

Is this only for handling the `oclif` internal logging scenario and not to be imported from (for example) plugins?  Or does this add native logging functions to every oclif-derived command so that sf commands don't need to instantiate Logger themselves?

Reply:
Primarily for internal logging in oclif/core but other plugins could access it from oclif/core if they wanted

It does not add native logging functions to every oclif-derived command